### DISCUSSION TITLE
GPS module red light staying on not blinking. NO GPS on screen.

### DISCUSSION THREAD

**Discussion by VPiREtro** [JosephHewitt/wardriver_rev3#153] [Q&A]
I'm using the battery atachment, was having trouble with the sim800l not getting enough power at first by powering through micro usb port with PC. after attaching battery, sim800l module now works but GPS seems to not be functioning. The red led stays on, also noticed the time is 5 hours off. Any input helps thanks!!

**Comment by pejacoby** [comment]
What build are you using?  I haven't seen that issue, using Anker 10K MaH battery packs.  Does the display show "No GPS" all the time?

**Comment by JosephHewitt** [comment]
Hi,

The light on the GPS module indicates it has power, and then it will start to blink once it has a location fix. It's usually not easy to get a fix from inside a building so you should take the device outside and ideally stand away from any buildings or tall structures. If you still do not get a GPS location after about 3 minutes of being outside away from buildings you can check the following:

- All of the GPS antennas I have tested are directional, this means you need to point the "top" towards the sky. On my antennas, the **bottom** is  the fully shiny metallic side, so that side should point **opposite** to the sky.
- The ESP modules interfere with the GPS antenna, so there needs to be a small gap between. Usually I make sure there is a distance of around 2cm (about 3/4 of an inch) between the antenna and the ESP modules. Usually I wrap my antenna in electrical tape and then glue it to the top of the SD card holder (again, be sure to point the correct side towards the sky). If you use the 3D printed enclosure with the antenna holder, the holder works just fine.
- Check the antenna is actually plugged in correctly.
- The antenna connector which plugs into the GPS module is able to rotate, but if you rotate it far enough it can come into contact with other metal on the GPS module (specifically part of the small battery) - this will cause your antenna to not receive signals so you should avoid it coming into contact like that.

If none of the above helps, try replacing the antenna and then replace the whole GPS module if it still doesn't work.

The time being off by 5 hours is normal because the time is displayed in the UTC timezone. We have an issue open for this at #50.

**Reply by VPiREtro** [reply]
Thanks y'all, and thanks Joseph for creating this awesome diy project. Turns  out it was a faulty esp32. I built a second one and it works great! Thank you for the input, good to know all that info. I appreciate y'all attempting to help!
